cmd/landscaper-controller/app/app.go

```diff
@@ -71,6 +77,84 @@ func (o *options) run(ctx context.Context) error {
 		return fmt.Errorf("unable to setup execution controller: %w", err)
 	}
 
+	///// WEBHOOK /////
+
+	// initialize webhook options
+	wo := webhook.Options{
+		WebhookConfigurationName: "landscaper-validation-webhook",
+		WebhookBasePath:          "/webhook/validate/",
+		WebhookNameSuffix:        ".validation.landscaper.gardener.cloud",
+		ObjectSelector: metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{
+				{
+					Operator: metav1.LabelSelectorOpNotIn,
+					Key:      "validation.landscaper.gardener.cloud/skip-validation",
+					Values:   []string{"true"},
+				},
+			},
+		},
+		ServicePort: o.webhookServicePort,
+	}
+	webhookLogger := ctrl.Log.WithName("webhook").WithName("validation")
+	webhookedResources := []webhook.WebhookedResourceDefinition{}
+	// add validation webhooks for Installations, DeployItems, and Executions (unless disabled)
+	if o.disabledWebhooks == nil || !o.disabledWebhooks["all"] {
+		if o.webhookNamespace == "" {
+			return errors.New("webhook service namespace must not be empty")
+		}
+		if o.webhookName == "" {
+			return errors.New("webhook service name must not be empty")
+		}
```

Comment:
i would separate the validation and the execution.
Lets create a dedicated validation func for the options `func (o *options) Validate() error`